### ISSUE TITLE
Improve URL parsing

### ISSUE THREAD

**Issue by IstoraMandiri** [twitter-together/action#221]
**What happened?**

After testing pull requests with various URLs passed to the `retweet` option, the preview check returned "valid", but after merging the tweet failed to published.

**What did you expect to happen?**

The preview should fail if the URL is not formed correctly, or the publish action should parse the URL properly before retweeting it.

**Additionally**

An alternative approach is to ignore URLs entirely, and / or accept the Tweet ID. 

```tweet
---
retweet: 1575820698877980672
---
```

### URL with Hash

https://github.com/IstoraMandiri/twitter-together-testing/pull/8

FAILING: https://github.com/IstoraMandiri/twitter-together-testing/actions/runs/3168209652

```tweet
---
retweet: https://twitter.com/testing_tt_/status/1576462993302372355#blah
---
```

### URL with Mock Tracker

https://github.com/IstoraMandiri/twitter-together-testing/pull/9

FAILING: https://github.com/IstoraMandiri/twitter-together-testing/actions/runs/3168219858

```tweet
---
retweet: https://twitter.com/testing_tt_/status/1576508832418873349?cxt=xxx
---
```

### URL with `mobile` subdomain

https://github.com/IstoraMandiri/twitter-together-testing/pull/9

FAILING: https://github.com/IstoraMandiri/twitter-together-testing/actions/runs/3168233222

```tweet
---
retweet: https://mobile.twitter.com/testing_tt_/status/1575820698877980672
---
```

**Comment by gr2m** [comment]
> An alternative approach is to ignore URLs entirely, and / or accept the Tweet ID.
> 
> ```
> ---
> retweet: 1575820698877980672
> ---
> ```

I'd prefer to have the full URL, it makes it easier to lookup the original tweet. But happy to be convinced otherwise. I see the benefit of only accepting IDs, and having a helpful error message in case an invalid value was provided.

**Comment by MattIPv4** [comment]
I think the issue here is really just that we don't validate the URLs during preview -- the regex matching only happens during the tweet process currently:

https://github.com/twitter-together/action/blob/66e7a8688eb6d0ef561dce4b7b43fd3d879d7625/lib/common/tweet.js#L16-L17

https://github.com/twitter-together/action/blob/66e7a8688eb6d0ef561dce4b7b43fd3d879d7625/lib/common/tweet.js#L38-L39

https://github.com/twitter-together/action/blob/66e7a8688eb6d0ef561dce4b7b43fd3d879d7625/lib/common/tweet.js#L47-L48

I think we probably just need to validate URLs during preview and throw errors if they don't match the expected pattern.

**Comment by gr2m** [comment]
yes I agree, let's do that, we can discuss changing what `retweet` can be set to afterwards
